rfc/rfc022-discovery-service.md

```diff
@@ -0,0 +1,299 @@
+
+Servers MUST share a JSON document describing the discovery service. This document is known as the _service definition_.
+The document MUST contain the following properties:
+- `id` REQUIRED. String containing a value that identifies the service definition. MAY be used to version the definition.
+- `did_methods` OPTIONAL. Array of strings containing the DID methods enabled for the service. If not present, all DID methods are allowed.
+- `endpoint` REQUIRED. String containing the URL where the server hosts the service endpoint.
+- `presentation_definition` REQUIRED. JSON object with the Presentation Definition (see [PEX]) describing requirements for presentations on the list.
+- `presentation_max_validity` REQUIRED. JSON number containing the maximum validity period (number of seconds between `nbf` and `exp`) of a presentation in seconds.
+
+For example:
+
+``json
+{
+  "id": "uc_university_v1",
+  "endpoint": "https://example.com/usecase/university/v1",
+  "presentation_max_validity": 259200,
+  "presentation_definition": {
+    "id": "pd_university",
+    "input_descriptors": [
+      {
+        "id": "pd_university_type",
+        "constraints": {
+          "fields": [
+            {
+              "path": ["$.type"],
+              "filter": {
+                "type": "string",
+                "const": "UniversityCredential"
+              }
+            },
+            {
+              "path": "$.credentialSubject.name",
+              "filter": {
+                "type": "string"
+              }
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+``
+
+## 6. Trust
+
+Trust of credential issuers (e.g. `did:example:education-accreditor` issuing `EducationalInstitutionCredential`) SHOULD be defined by the presentation definition.
+In this case, there should be 2 constraints: one for the type and one for the issuer:
+
+``json
+[
+  {
+    "path": ["$.type"],
+    "filter": {
+      "type": "string",
+      "const": "EducationalInstitutionCredential"
+    }
+  },
+  {
+    "path": "$.issuer",
+    "filter": {
+      "type": "string",
+      "const": "did:example:education-accreditor"
+    }
+  }
+]
+``
+
+## 7. Security Considerations
+
+## 7.1 Denial of Service
+
+Since the server is hosting a public endpoint, it should take the following measures to prevent denial-of-service attacks:
+- rate limiting on the number of requests per client.
+- rate limiting on the number of requests per second.
+- limit the size of the list.
+- limit the size of the presentations that can be uploaded.
+
+Clients should also take measures when the server has been compromised and is hosting malicious content:
+- limit the number of presentations to process.
+- limit the size of the presentations to process.
+- use streaming processing to prevent excessive memory usage.
+
+## 7.2 Trust
+
+Designers of services should build trust into the Presentation Definition of the server. This can be done by specifying specific issuers of credentials.
```

Comment:
```suggestion
Designers of services should write trust into the Presentation Definition of the Service Definition. This can be done by specifying specific issuers of credentials.
```